src/xrUICore/XML/UITextureMaster.cpp

```diff
@@ -54,6 +56,8 @@ void CUITextureMaster::ParseShTexInfo(pcstr path, pcstr xml_file)
             /* avo: fix issue when values were not updated (silently skipped) when same key is encountered more than once. This is how std::map is designed. 
             /* Also used more efficient C++11 std::map::emplace method instead of outdated std::pair::make_pair */
             /* XXX: avo: note that xxx.insert(mk_pair(v1,v2)) pattern is used extensively throughout solution so there is a good potential for other bug fixes/improvements */
+            static Lock new_texture_lock;
```

Comment:
Im think that this place dont needed any sync, because in Windows no problem with textures. Most likely, the problem is elsewhere, because the game on linux immediately drops, and in different places.